src/features/bundlr/BundlrInput.tsx

```diff
@@ -4,35 +4,89 @@ import {
   NumberInputStepper,
   NumberIncrementStepper,
   NumberDecrementStepper,
-  NumberInputProps,
+  InputGroup,
+  InputRightElement,
 } from '@chakra-ui/react';
+import { ethers } from 'ethers';
+import { uploadPriceDecimals } from 'lib/constants';
+import { useState } from 'react';
 
-export function BundlrInput(props: NumberInputProps) {
-  return (
-    <NumberInput
-      step={0.01}
-      mt={3}
-      w="100%"
-      min={0}
-      color="whiteAlpha.700"
-      allowMouseWheel
-      {...props}
-    >
-      <NumberInputField
-        borderColor="whiteAlpha.700"
-        color="whiteAlpha.800"
-        h="48px"
-      />
-      <NumberInputStepper>
-        <NumberIncrementStepper
-          color="whiteAlpha.700"
-          borderColor="whiteAlpha.700"
-        />
-        <NumberDecrementStepper
-          color="whiteAlpha.700"
+const inputLimit = 46;
+
+export function useBundlrInput(args: { initialAmount?: number; inputHeight?: string } = {}) {
```

Comment:
The pattern to use a hook that returns a component is a legitimate pattern, but I'm curious what made you decide to go with turning this into a hook instead of leaving it as a component.